### PR TITLE
MCN-66 | Add and refactor tests that are not included in our suite

### PR DESCRIPTION
**Pull request by cmartinezledo**

### CODE REVIEW

File: src/test/java/org/mule/module/mongo/MongoClientConnectivityTestCases.java [hunk 1]
```diff
@@ -11,19 +11,40 @@
...
+    private String pass;
+    private String db;
+    private String host;
+    private String port;
 
     @Before
-    public void setup() {
+    public void setup() throws IOException {
+        final Properties prop = new Properties();
+        try (final InputStream input = getClass().getResourceAsStream("/automation-credentials.properties")) {
```
Comment (urbanusjam) [inline src/test/java/org/mule/module/mongo/MongoClientConnectivityTestCases.java]: Try using a *ConfigurationUtils* class to load the properties:

@Before
  public void setup(){
    validCredentials = ConfigurationUtils.getAutomationCredentialsProperties();
  }

More info here: http://mulesoft.github.io/connector-certification-docs/advanced/index.html#_system_tests
Download link: http://mulesoft.github.io/connector-certification-docs/advanced/files/ConfigurationUtils.java

File: src/test/java/org/mule/module/mongo/automation/testcases/FindObjectsTestCases.java [hunk 2]
```diff
@@ -48,9 +48,9 @@ public void testFindObjects() {
...
         for (Document obj : payload) {
             String dbObjectID = obj.get("_id").toString();
-            assertTrue(objectIDs.contains(dbObjectID));
+            assertTrue("objectIDs does not contain: " + obj, objectIDs.contains(dbObjectID));
         }
-        assertTrue(objectIDs.size() == Iterables.size(payload));
+        assertThat(objectIDs.size(), is(Iterables.size(payload)));
     }
 
     @Category({ RegressionTests.class })
```
Comment (urbanusjam) [inline src/test/java/org/mule/module/mongo/automation/testcases/FindObjectsTestCases.java]: Categories are no longer needed.

File: src/test/java/org/mule/module/mongo/MongoClientConnectivityTestCases.java [hunk 3]
```diff
@@ -34,40 +55,43 @@ public void tearDown() {
...
+    @Test
+    public void connectionIncorrectDatabase() {
+        assertFalse(isConnected(user, pass, "zdrgdr"));
     }
 
     @Test
     public void connectionIncorrectCredentials() {
-        config.setPort("27017");
-        assertFalse(isConnected("admin", "zdrgdr", "test"));
+        assertFalse(isConnected(user, "zdrgdr", db));
```
Comment (urbanusjam) [inline src/test/java/org/mule/module/mongo/MongoClientConnectivityTestCases.java]: assertThat(isConnected(user, "zdrgdr", db), is(false));

File: src/test/java/org/mule/module/mongo/MongoClientConnectivityTestCases.java [hunk 4]
```diff
@@ -34,40 +55,43 @@ public void tearDown() {
...
     }
 
     @Test
     public void validConnectionReplicaSet() {
-        config.setPort("27017, 27017");
-        config.setHost("127.0.0.1, 127.0.0.1");
-        assertTrue(isConnected("admin", "", "test"));
+        config.setPort(port + ", " + port);
+        config.setHost(host + ", " + host);
+        assertTrue(isConnected(user, pass, db));
```
Comment (martinpaoloni) [inline src/test/java/org/mule/module/mongo/MongoClientConnectivityTestCases.java]:
```java
assertThat(isConnected(user, pass, db), is(true));
```

File: src/test/java/org/mule/module/mongo/automation/testcases/FindObjectsTestCases.java [hunk 5]
```diff
@@ -48,9 +48,9 @@ public void testFindObjects() {
 
         for (Document obj : payload) {
             String dbObjectID = obj.get("_id").toString();
-            assertTrue(objectIDs.contains(dbObjectID));
+            assertTrue("objectIDs does not contain: " + obj, objectIDs.contains(dbObjectID));
```
Comment (martinpaoloni) [inline src/test/java/org/mule/module/mongo/automation/testcases/FindObjectsTestCases.java]: Change all those ```assertTrue``` to ```assertThat(..., is(true))``` ;-)

File: src/test/java/org/mule/module/mongo/automation/util/ConfigurationUtils.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+        try {
+            ret.load(new FileInputStream(automationPath));
+        } catch (IOException e) {
+            throw new ConfigurationLoadingFailedException("Can not load Automation Credentials File within path " + automationPath, e);
+        }
+
+        return ret;
+    }
+}
```
Comment (urbanusjam) [inline src/test/java/org/mule/module/mongo/automation/util/ConfigurationUtils.java]: Add new empty line at the end of the file.

**Comment by martinpaoloni** [comment]
+1

**Pull request merged**